src/sdk/ui/modal/Provider.tsx

```diff
@@ -0,0 +1,56 @@
+import React, {
+  useMemo,
+  useState,
+  useContext,
+  useCallback,
+  createContext,
+} from 'react'
+import type { PropsWithChildren } from 'react'
+
+import './provider.scss'
```

Comment:
hmm, I think it's better to identify with the prefix otherwise sounds too generic, how about renaming to `modal-provider.scss`?